xclim/core/base.py

```diff
@@ -0,0 +1,407 @@
+            self.validate(da)
+        self.cfprobe(*das.values())
+
+        # Compute the indicator values, ignoring NaNs.
+        out = self.compute(**das, **ba.kwargs)
+
+        # Convert to output units
+        out = convert_units_to(out, self.units, self.context)
+
+        # Update netCDF attributes
+        out.attrs.update(attrs)
+
+        # Bind call arguments to the `missing` function, whose signature might be different from `compute`.
+        mba = signature(self.missing).bind(*das.values(), **ba.arguments)
+
+        # Mask results that do not meet criteria defined by the `missing` method.
+        mask = self.missing(*mba.args, **mba.kwargs)
+        ma_out = out.where(~mask)
+
+        return ma_out.rename(vname)
+
+    def update_attrs(self, ba, das):
+        """Update attributes.
+
+        Parameters
+        ----------
+        ba: bound argument object
+          ...
+        das: tuple
+          Input arrays.
+        """
+        args = ba.arguments
+        out = self.format(self.cf_attrs, args)
+        for locale in LOCALES:
+            out.update(
+                self.format(
+                    get_local_attrs(
+                        self,
+                        locale,
+                        names=self._cf_names,
+                        fill_missing=False,
+                        append_locale_name=True,
+                    ),
+                    args=args,
+                    formatter=get_local_formatter(locale),
+                )
+            )
+
+        out["var_name"] = vname = self.format({"var_name": self.var_name}, args)[
+            "var_name"
+        ]
+
+        # Update the signature with the values of the actual call.
+        cp = OrderedDict()
+        for (k, v) in ba.signature.parameters.items():
+            if v.default is not None and isinstance(v.default, (float, int, str)):
+                cp[k] = v.replace(default=ba.arguments[k])
+            else:
+                cp[k] = v
+
+        # Get history and cell method attributes from source data
+        attrs = defaultdict(str)
+        attrs["cell_methods"] = merge_attributes(
+            "cell_methods", new_line=" ", missing_str=None, **das
+        )
+        if "cell_methods" in out:
+            attrs["cell_methods"] += " " + out.pop("cell_methods")
+
+        attrs["history"] = update_history(
+            f"{self.identifier}{ba.signature.replace(parameters=cp.values())}",
+            new_name=vname,
+            **das,
+        )
+
+        attrs.update(out)
+        return attrs
+
+    @staticmethod
+    def check_identifier(identifier):
+        """Verify that the identifier is a proper slug."""
+        if not re.match(r"^[-\w]+$", identifier):
+            warnings.warn(
+                "The identifier contains non-alphanumeric characters. It could make life "
+                "difficult for downstream software reusing this class.",
+                UserWarning,
+            )
```

Comment:
I think I would be in favor of being stricter about this. What does @Zeitsperre think?

Reply:
Yes, this could cause bigger problems down-the-line. Until we explicitly support things like emojis, let's not let them be passed along.